lib/fizzy/parser.cpp

```diff
@@ -269,7 +269,8 @@ parser_result<std::string> parse_string(const uint8_t* pos, const uint8_t* end)
     uint32_t size;
     std::tie(size, pos) = leb128u_decode<uint32_t>(pos, end);
 
-    if ((end - pos) < size)
+    assert(pos <= end);
+    if (static_cast<size_t>(end - pos) < size)
```

Comment:
What does the cast solve?

Reply:
Pointer diff is int, size_t is unsigned int.